Create 1.26.1 documentation (#358)
canonical/anbox-cloud-docs: reference/release-notes/1.26.1.md

Anbox Cloud 1.26.1 is a patch release. To understand minor and patch releases, see [Release notes](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/release-notes/release-notes).

Please see [Component versions](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/component-versions/) for a list of updated components.

## Requirements

See the [Requirements](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/requirements/) for details on general and deployment specific requirements to run Anbox Cloud.

## New features & improvements
* The [HAProxy charm](https://charmhub.io/haproxy) now is also deployed on Ubuntu 22.04. This helps us offer the charm with the same Ubuntu version as the other charms that are part of the Anbox Cloud bundle.<!--AC-3329-->
* Android security updates for June 2025 (see [Android Security Bulletin - June 2025](https://source.android.com/docs/security/bulletin/2025-06-01)).
* The Android WebView has been updated to [136.0.7103.60](https://chromereleases.googleblog.com/2025/04/chrome-for-android-update_29.html).<!--AC-3316-->

## Deprecations

TBD

## Known issues

> 

Author: Any known issues?

Reviewer: https://launchpad.net/anbox-cloud/+milestone/1.26.2

To me, we would have to mention
- 2111598
- 2112098

I'll leave @morphis comment all others

---
orphan: true
---
# 1.26.1

These release notes cover new features and changes in Anbox Cloud 1.26.1.

> Anbox Cloud 1.26.1 is a minor release. To understand minor and patch releases, see [Release notes](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/release-notes/release-notes).

Reviewer: Anbox Cloud 1.26.1 is a minor release.
->
Anbox Cloud 1.26.1 is a patch release.

* Android security updates for June 2025 (see [Android Security Bulletin - June 2025](tbd)).
* The Android WebView has been updated to [137.0.7151.61](https://chromereleases.googleblog.com/2025/05/chrome-for-android-update_0122535239.html).<!--AC-3316-->

## Deprecations

TBD

## Known issues

TBD

## CVEs

This release includes fixes from the respective upstreams and components for the following CVEs:

| CVE | Affected Components |
|-----|---------------------|

## Bug fixes

The following bugs were fixed as part of the 1.26.1 release:

* [LP 2106113](https://bugs.launchpad.net/anbox-cloud/+bug/2106113) There were two commands in AMC that looked similar but had different purposes - `show-log` and `logs`. This was confusing about when to use these commands. The help text is now clarified to facilitate better understanding that `show-logs` is to obtain logs from an instance that is stopped or in an error state while `logs` is to obtain logs from a started or a running instance.<!--AC-3293-->

> * [LP 2112098](https://bugs.launchpad.net/anbox-cloud/+bug/2112098) While upgrading to 1.26.0 from 1.25.2, the dashboard displays the error `Ubuntu Pro token is not attached` even when the Ubuntu Pro token is already attached.

Reviewer: As discussed in the SU, the bug fix has been postponed to 1.26.2.

## Deprecations

TBD

## Known issues

TBD

## CVEs

This release includes fixes from the respective upstreams and components for the following CVEs:

| CVE | Affected Components |
|-----|---------------------|

## Bug fixes

The following bugs were fixed as part of the 1.26.1 release:

* [LP 2106113](https://bugs.launchpad.net/anbox-cloud/+bug/2106113) There were two commands in AMC that looked similar but had different purposes - `show-log` and `logs`. This was confusing about when to use these commands. The help text is now clarified to facilitate better understanding that `show-logs` is to obtain logs from an instance that is stopped or in an error state while `logs` is to obtain logs from a started or a running instance.<!--AC-3293-->
* [LP 2112098](https://bugs.launchpad.net/anbox-cloud/+bug/2112098) While upgrading to 1.26.0 from 1.25.2, the dashboard displays the error `Ubuntu Pro token is not attached` even when the Ubuntu Pro token is already attached.
* [LP 2111893](https://bugs.launchpad.net/anbox-cloud/+bug/2111893) Stream can not be initiated due to intermittent TLS handshake failures.

> * [LP 2111598](https://bugs.launchpad.net/anbox-cloud/+bug/2111598) When the developer settings are enabled, trying to access the *System* section of the *Settings* app in Android 15 causes a native crash.

Reviewer 2: This bug was moved to 1.26.2
```suggestion
```

* [LP 2100933](https://bugs.launchpad.net/anbox-cloud/+bug/2100933) AMS HTTP API documentation was missing information regarding the `/1.0/registry` endpoints.
* [LP 2087828](https://bugs.launchpad.net/anbox-cloud/+bug/2087828) This bug is related to [LP #2086814](https://bugs.launchpad.net/bugs/2086814) which is about the Stream Gateway creating multiple sessions for the same container when in a HA deployment.
However, after restarting the Stream Agent, all of the sessions were terminated, leaving the instance in a state where you cannot stream from it.

> * [LP 2097511](https://bugs.launchpad.net/anbox-cloud/+bug/2097511) The `minidumper` thread in the Anbox runtime is responsible for handling crashes in the Anbox runtime.When [profiling](https://pprof.me/9d7c55bbddc28584dc319c6d5c96dfe1), this thread was seen consuming excess CPU time even when a crash had not been detected yet.<!--AC-3330-->

Reviewer 2: ```suggestion
* [LP 2097511](https://bugs.launchpad.net/anbox-cloud/+bug/2097511) The `minidumper` thread in the Anbox runtime is responsible for handling crashes in the Anbox runtime. When [profiling](https://pprof.me/9d7c55bbddc28584dc319c6d5c96dfe1), this thread was seen consuming excess CPU time even when a crash had not been detected yet.<!--AC-3330-->
```

Also, I'm not sure if we should link to actual profiling data, as I don't think users are supposed to have access to our debug symbols. @morphis Any thoughts in this?

Author: I wondered about that too but the link is publicly available, so I thought it was okay to include it.

Reviewer 3: @ajanon there are no debug symbols involved, just what is in the images we ship already today, so nothing is beign disclosed.

Reviewer 3: Essentially everyone having access to Anbox Cloud can create the same today already.

Reviewer 2: The function names are fully disclosed here. I'm not sure how important that is, but it's definitely in plain sight in that link.

Author: After discussing in SU, we decided to drop the link as although it does not have any private information, for the current context, it does not add a lot of value when compared to the concerns about the function names being disclosed.

* [LP 2106104](https://bugs.launchpad.net/anbox-cloud/+bug/2106104) The appliance initialization fails on a new machine that has an NVIDIA GPU but the driver is not yet installed. You see the following error when you try to initialize the appliance:<!--AC-3332-->

        Error: failed to parse NVIDIA driver version: strconv.ParseUint: parsing "": invalid syntax
* [LP 2111486](https://bugs.launchpad.net/anbox-cloud/+bug/2111486) The LXD charm was updated recently to avoid queries to the snap store to get the version information when the snap is already installed. Instead, the charm retrieves the necessary snap information using `snap list`.
However, the lookup sometimes fails and falls back to querying the snap store. This results in timeout exceptions from time to time, especially when a snap proxy is configured behind the store.<!--AC-3364-->

> * [LP 2106114](https://bugs.launchpad.net/anbox-cloud/+bug/2106114) When you attempt to rotate the screen from portrait to landscape during Android runtime, there is no error observed but the screen does not rotate for AOSP images. It works as expected for AAOS images, though.<!--AC-3370-->

Reviewer 2: Slightly more accurate description IMO:
```suggestion
* [LP 2106114](https://bugs.launchpad.net/anbox-cloud/+bug/2106114) When you attempt to rotate the screen from portrait to landscape by setting the `user_rotation` setting using `adb`, there is no error observed but the screen does not rotate for AOSP images. It works as expected for AAOS images, though.<!--AC-3370-->
```

The bug is pretty specific and only concerns setting the rotation this way, through an adb command. Other ways of setting the rotation should work as expected.

Please see [Component versions](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/component-versions/) for a list of updated components.

## Requirements

See the [Requirements](https://documentation.ubuntu.com/anbox-cloud/en/latest/reference/requirements/) for details on general and deployment specific requirements to run Anbox Cloud.

## New features & improvements
* The [HAProxy charm](https://charmhub.io/haproxy) now is also deployed on Ubuntu 22.04. This helps us offer the charm with the same Ubuntu version as the other charms that are part of the Anbox Cloud bundle.<!--AC-3329-->
* Android security updates for June 2025 (see [Android Security Bulletin - June 2025](https://source.android.com/docs/security/bulletin/2025-06-01)).
* The Android WebView has been updated to [136.0.7103.60](https://chromereleases.googleblog.com/2025/04/chrome-for-android-update_29.html).<!--AC-3316-->

## Known issues

The following known issues exist in Anbox Cloud and are planned to be fixed in subsequent releases:

> * [LP 2113771](https://bugs.launchpad.net/anbox-cloud/+bug/2113771) When upgrading charms from 1.26.0 to 1.26.1, the AMS node controller ends up in an error state. You don't have to worry about this as the AMS node controller is deprecated since 1.26.0 and the port forwarding is instead done by LXD.

Reviewer: To work around this:
1. Check if the `exposed_instance_ports` config is set to a specific port or port range for lxd charm:
       $ juju config lxd exposed_instance_ports
    If it is set, reset the configuration:
       $ juju config lxd --reset exposed_instance_ports
2. After resetting the exposed_instance_ports config for each LXD unit (you can verify by checking the `Ports` column in the output of juju status), temporarily open the required ports on the AMS node controller:
       $ juju exec -a ams-node-controller -- open-port 10000-11001/TCP
3. Upgrade the ams node controller charm to 1.26/stable channel
   $ juju refresh ams-node-controller --channel=1.26/stable

With this, the AMS node controller will upgrade without any issues.

NOTE: This issue only affects the Anbox Cloud patch release upgrade from 1.26.0 to 1.26.1. If you're upgrading from the previous minor release (e.g., 1.25), you can skip the above workaround.
